_docs/Qabel-Protocol-Box.md

```diff
@@ -0,0 +1,184 @@
+---
+title: "Protocol: box"
+---
+# Box Specification
+
+## Abstract
+
+A set of protocols to store files and folders on a VOLUME that is on S3 and managed by a third party who doesn't need to be trusted.
+
+## Used services
+
+Qabel Box uses an Accounting server which controls the access to S3. Every client who needs write access has to be authenticated by the server and then receives a set of credentials that he can use for direct access to the VOLUME.
+Qabel Box also directly uses an S3 compatible server to store the blocks and metadata.
+
+## Structure of a VOLUME
+
+A Volume consists of metadata files and blocks. Blocks have a block size of up to 10mb. Every VOLUME has a metadata file at VOLUME/index which is the starting point and contains references to other objects.
+
+All mtime values are seconds since epoch in UTC.
+
+The metadata file stores information equivalent of this example JSON document, but stored in an SQLite database (the database schema is explained later):
+
+```JSON
+{
+path: "/users/b5911736-9ace-a799-8e34-dd9c17acff9a/",
+name: "index",
+version: 7,
+objects: [
+{ name: "foobar.jpg", type: "file", size: 6203434, mtime: 1445432325, blocks: [
```

Comment:
What about access rights? Should we add a key-value-map for later introduction of additonal file attributes like ACL to the spec at this stage? Or at least make a note?